client/src/components/molecules/JobPosting.js

```diff
@@ -31,15 +32,42 @@ const JobPosting = (props) => {
     details.pay,
     details.candidates,
   ];
-  
-  data.matches = useSelector((state) => state.matches.page3Object?.page3Students?.length);
+
+  data.matches = useSelector(
```

Comment:
we should keep our redux state access persistent, there is no need to use useSelector because you can access your state via your mapStateToProps() function at the bottom of this page. Please review and update this section and any other section you use useSelector()